nmmo/core/env.py

```diff
@@ -408,20 +409,18 @@ def _compute_rewards(self, agents: List[AgentID], dones: Dict[AgentID, bool]):
           entity identified by ent_id.
     '''
     # Initialization
-    infos = {}
-    for agent_id in agents:
-      infos[agent_id] = {}
-      infos[agent_id]['task'] = {}
-    rewards = {agent_id: 0 for agent_id in agents}
+    infos = {agent_id: {'task': {}} for agent_id in agents}
+    rewards = defaultdict(int)
 
     # Compute Rewards and infos
-    self.game_state = self._gamestate_generator.generate(self.realm, self.obs)
-    for task in self.tasks:
-      task_rewards, task_infos = task.compute_rewards(self.game_state)
-      for agent_id, reward in task_rewards.items():
-        if agent_id in agents and agent_id not in dones:
-          rewards[agent_id] = rewards.get(agent_id,0) + reward
-          infos[agent_id]['task'][task.name] = task_infos[agent_id] # progress
+    if agents and not all(dones.values()):
```

Comment:
dones only contains dead agents, so all(dones) becomes true on any turn that an agent dies. we should have a test for this, since it's an easy error to make